File: Backend/src/main/java/com/frozenleafstudio/dev/AutomatedSetlist/Artist/Artist.java
package com.frozenleafstudio.dev.AutomatedSetlist.Artist;

import java.util.List;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import com.frozenleafstudio.dev.AutomatedSetlist.Setlist.Setlist;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "Artists")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Artist {
    @Id
    private ObjectId id;
    private String mbid; // Musicbrainz Identifier
    private String tmid; // Ticket Master Identifier
    private String name;
    private String sortName;
    private String disambiguation;
    private String url;
    @DBRef(lazy = true)
    private List<Setlist> setlists;
}

File: Frontend/src/models/Artist.ts
import { Setlist } from "./Setlist";

export class Artist {
  id: string = "";
  mbid: string = ""; // Musicbrainz Identifier
  tmid?: string = ""; // Ticket Master Identifier
  name: string = "";
  sortName: string = "";
  disambiguation?: string; // Optional, represents artist genre
  url: string = "";
  setlists?: Setlist[]; // Optional, initially not populated

  constructor(initializer: Partial<Artist> = {}) {
    Object.assign(this, initializer);
  }
}
